lib/asn1/der_copy.c

```diff
@@ -99,11 +99,13 @@ int ASN1CALL
 der_copy_printable_string (const heim_printable_string *from,
 		       heim_printable_string *to)
 {
-    to->length = from->length;
-    to->data   = malloc(to->length + 1);
-    if(to->data == NULL)
+    assert(from->length == 0 || (from->length > 0 && from->data != NULL));
+    to->data = malloc(from->length + 1);
+    if (to->data == NULL)
 	return ENOMEM;
```

Comment:
Do we want to initialize `to->length` in the `ENOMEM` case?

Reply:
I've added setting length to zero at all error returns in the next version of the patch.